include/sqlpp11/detail/parse_date_time.h

```diff
@@ -27,102 +27,237 @@
+        input = new_input;
+        return true;
       }
-      value = ::sqlpp::chrono::day_point{
-        ::date::year{std::atoi(date_string + mr.position(1))} / // Year
-        std::atoi(date_string + mr.position(2)) /               // Month
-        std::atoi(date_string + mr.position(3))                 // Day of month
-      };
-      return true;
-    }
 
-    // Parse a date string formatted as YYYY-MM-DD HH:MM:SS.US TZ
-    // .US are optional fractional seconds, up to 6 digits in length
-    // TZ is an optional time zone offset formatted as +HH[:MM] or -HH[:MM]
-    //
-    inline bool parse_string_date_time(::sqlpp::chrono::microsecond_point& value, const char* date_time_string)
-    {
-      static const std::regex rx{
-        "(\\d{4})-(\\d{2})-(\\d{2}) "
-        "(\\d{2}):(\\d{2}):(\\d{2})(?:\\.(\\d{1,6}))?"
-        "(?:([+-])(\\d{2})(?::(\\d{2})(?::(\\d{2}))?)?)?"
-      };
-      std::cmatch mr;
-      if (std::regex_match(date_time_string, mr, rx) == false)
+      inline bool parse_character(const char*& input, char ch)
       {
-        return false;
+        if (*input != ch)
+        {
+          return false;
+        }
+        ++input;
+        return true;
       }
-      value =
-        ::sqlpp::chrono::day_point{
-          ::date::year{std::atoi(date_time_string + mr.position(1))} / // Year
-          std::atoi(date_time_string + mr.position(2)) /               // Month
-          std::atoi(date_time_string + mr.position(3))                 // Day of month
-        } +
-        std::chrono::hours{std::atoi(date_time_string + mr.position(4))} +     // Hour
-        std::chrono::minutes{std::atoi(date_time_string + mr.position(5))} +   // Minute
-        std::chrono::seconds{std::atoi(date_time_string + mr.position(6))} +   // Second
-        ::std::chrono::microseconds{                                           // Second fraction
-          mr[7].matched ? std::stoi((mr[7].str() + "000000").substr(0, 6)) : 0
-        };
-      if (mr[8].matched)
+
+      inline bool parse_yyyy_mm_dd(sqlpp::chrono::day_point& dp, const char*& input)
       {
-        const auto tz_sign = (date_time_string[mr.position(8)] == '+') ? 1 : -1;
-        const auto tz_offset =
-          std::chrono::hours{std::atoi(date_time_string + mr.position(9))} +
-          std::chrono::minutes{mr[10].matched ? std::atoi(date_time_string + mr.position(10)) : 0} +
-          std::chrono::seconds{mr[11].matched ? std::atoi(date_time_string + mr.position(11)) : 0};
-        value -= tz_sign * tz_offset;
+        auto new_input = input;
+        int year, month, day;
+        if ((parse_unsigned(year, new_input, 4) == false) || (parse_character(new_input, '-') == false) ||
+            (parse_unsigned(month, new_input, 2) == false) || (parse_character(new_input, '-') == false) ||
+            (parse_unsigned(day, new_input, 2) == false))
+        {
+          return false;
+        }
+        const auto ymd = ::date::year{year} / month / day;
```

Comment:
As we don't do the check, we would drop this temporary variable.

Reply:
That is a good catch. The variable was a remnant from the old version of the code which had a call to `ymd.ok()`. I removed the temporary variable but still added a call to a custom function that checks the validity of the year/month/day.